rewrite-java/src/main/java/org/openrewrite/java/utils/ExpressionUtils.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.utils;
+
+import org.intellij.lang.annotations.Language;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Space;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+
+public final class ExpressionUtils {
```

Comment:
Since this class for now deals with string concatenation specifically, I would suggest to either rename the whole class or the individual utility methods. Because if someone were to use it to add two primitive ints for example, the returned `J.Binary` would be incorrectly type-attributed, which can have hard to track down undesirable consequences, I think.

Also, with this approach, it would for the given reason also be problematic to match the other `StringBuilder#append()` overloads.

Possibly this class should also be in an `internal` package?

Reply:
Moved this Util class to be package-private.